[Bug] Config composition error with latest version of OmegaConf

# 🐛 Bug
## Description

When using OmegaConf at commit 2dd15f9 (first commit where this problem occurs), there are multiple Hydra tests failures, for instance:

```
pytest "tests/test_basic_launcher.py::TestBasicLauncher::test_sweep_1_job[basic-overrides0]"
(...)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = __INVALID__, value = None

    def validate_and_convert(self, value: Any) -> Any:
        """
        Validates input and converts to canonical form
        :param value: input value
        :return: converted value ("100" may be converted to 100 for example)
        """
        if value is None:
            if self._is_optional():
                return None
>           raise ValidationError("Non optional field cannot be assigned None")
E           hydra.errors.ConfigCompositionException

../omegaconf/omegaconf/nodes.py:55: ConfigCompositionException
```

## Checklist
- [X] I checked on the latest version of Hydra
- [X] I created a minimal repro (See [this](https://stackoverflow.com/help/minimal-reproducible-example) for tips).

## To reproduce

Use master branch of Hydra with OmegaConf's commit 2dd15f9

## Additional context

This might actually be an OmegaConf bug (I'm not sure).

--- a/hydra/conf/__init__.py
+++ b/hydra/conf/__init__.py
@@ -95,7 +95,8 @@ class RuntimeConf:
     config_sources: List[ConfigSourceInfo] = MISSING
 
     # Composition choices dictionary
-    choices: Dict[str, str] = field(default_factory=lambda: {})
+    # Ideally, the value type would be Union[str, List[str], None]
+    choices: Dict[str, Any] = field(default_factory=lambda: {})
 
 
 @dataclass

